<launch>
  <arg name="debug" default="true" /> <!-- set to true to view the image or false to launch just the driver of the camera -->
  <arg name="num_cam" default="2" /> <!-- number of cameras, 1 or 2-->
  "$(eval arg('debug') ==0)"
  <node pkg="flir_module_driver" type="flir_module_driver_node" name="flir_module_driver" output="screen" respawn="false" if="$(eval arg('num_cam') == 1)">
    <param name="min_range_meas" type="int" value="30000" /> <!-- minimum measurement range for scaling to pixel value [0,255]-->
    <param name="max_range_meas" type="int" value="30960" /> <!-- maximum measurement range for scaling to pixel value [0,255]-->
    <param name="ir_img_color" type="bool" value="false" /> <!-- set to true to publish ir temp-coded color image, false for grayscale -->
    <param name="auto_gain" type="bool" value="false" /> <!-- set to true to autoadjust the scale instead of using a fix measurement range -->
    <param name="serial_number" type="string" value="0007003e-5119-3038-3532-373600000000" /> <!-- serial number of the device. On a console type: smesg | grep SerialNumber -->
  </node>

  <node pkg="flir_module_driver" type="flir_module_driver_node" name="flir_module_driver1" output="screen" respawn="false" if="$(eval arg('num_cam') == 2)">
    <param name="min_range_meas" type="int" value="30000" /> <!-- minimum measurement range for scaling to pixel value [0,255]-->
    <param name="max_range_meas" type="int" value="30960" /> <!-- maximum measurement range for scaling to pixel value [0,255]-->
    <param name="ir_img_color" type="bool" value="false" /> <!-- set to true to publish ir temp-coded color image, false for grayscale -->
    <param name="auto_gain" type="bool" value="false" /> <!-- set to true to autoadjust the scale instead of using a fix measurement range -->
    <param name="serial_number" type="string" value="80060001-5113-3238-3233-393800000000" /> <!-- serial number of the device. On a console type: smesg | grep SerialNumber -->
  </node>

  <node pkg="flir_module_driver" type="flir_module_driver_node" name="flir_module_driver2" output="screen" respawn="false" if="$(eval arg('num_cam') == 2)">
    <param name="min_range_meas" type="int" value="30000" /> <!-- minimum measurement range for scaling to pixel value [0,255]-->
    <param name="max_range_meas" type="int" value="30960" /> <!-- maximum measurement range for scaling to pixel value [0,255]-->
    <param name="ir_img_color" type="bool" value="false" /> <!-- set to true to publish ir temp-coded color image, false for grayscale -->
    <param name="auto_gain" type="bool" value="false" /> <!-- set to true to autoadjust the scale instead of using a fix measurement range -->
    <param name="serial_number" type="string" value="000c0035-510f-3538-3530-333500000000" /> <!-- serial number of the device. On a console type: smesg | grep SerialNumber -->
  </node>

  <!-- VISUALIZATION -->
  <group if="$(eval arg('debug') == 1)">
    <node name="rqt_image_view_thermal" pkg="rqt_image_view" type="rqt_image_view" respawn="true" output="screen" if="$(eval arg('num_cam') == 1)">
      <remap from="image" to="/flir_one_node/thermal/image_raw" />
    </node>
    <node name="rqt_image_view_thermal1" pkg="rqt_image_view" type="rqt_image_view" respawn="true" output="screen" if="$(eval arg('num_cam') == 2)">
      <remap from="image" to="/flir_one_node/thermal1/image_raw" />
    </node>
    <node name="rqt_image_view_thermal2" pkg="rqt_image_view" type="rqt_image_view" respawn="true" output="screen" if="$(eval arg('num_cam') == 2)">
      <remap from="image" to="/flir_one_node/thermal2/image_raw" />
    </node>
  </group>

</launch>
